build_module.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+# vim: ts=4 sw=4 expandtab
+#
+# Author: Feras Daoud <ferasda@mellanox.com>
+#
+
+ex()
+{
+    if ! eval "$@"; then
+        echo "Failed to execute: $@" >&2
+        exit 1
+    fi
+}
+
+tmpdir=`mktemp -d /tmp/nv.XXXXXX`
+if [ ! -d "$tmpdir" ]; then
+    echo "Failed to create a temp directory!" >&2
+    exit 1
+fi
+
+dirname=`basename "$PWD"`
+VERSION=`grep Version: *.spec | cut -d : -f 2 | sed -e 's@\s@@g'`
+RELEASE=`grep "define _release" *.spec | cut -d" " -f"4"| sed -r -e 's/}//'`
+if [ "X$VERSION" == "X" ] || [ "X$RELEASE" == "X" ]; then
+    echo "Failed to get version numbers!" >&2
+    exit 1
+fi
+
+ex cp -r . $tmpdir/nvidia_peer_memory-$VERSION
+pushd $tmpdir > /dev/null
+ex tar czf nvidia_peer_memory-$VERSION.tar.gz nvidia_peer_memory-$VERSION --exclude=.* --exclude=build_release.sh
+popd > /dev/null
+
+echo
+echo "Building source rpm for nvidia_peer_memory..."
+mkdir -p $tmpdir/topdir/{SRPMS,RPMS,SPECS,BUILD}
+ex "rpmbuild -ts --nodeps --define '_topdir $tmpdir/topdir' --define 'dist %{nil}' --define '_source_filedigest_algorithm md5' --define '_binary_filedigest_algorithm md5' $tmpdir/nvidia_peer_memory-$VERSION.tar.gz >/dev/null"
```

Comment:
this script might be invoked on a Debian based OS. 
Can't you simply print out this as the recipe for RPM distros ?